src/cli/java/org/commcare/util/screen/MenuScreen.java

```diff
@@ -34,13 +34,24 @@
     private String mTitle;
 
     public String[] getBadges() {
+        if (badges == null) {
+            calculateBadges();
+        }
         return badges;
     }
 
     public void setBadges(String[] badges) {
         this.badges = badges;
     }
 
+    private void calculateBadges() {
+        badges = new String[mChoices.length];
```

Comment:
I don't have a clue if we have a convention about this, but I think using `this.memberVar` instead of `memberVar` is more readable.